base/regex.jl

```diff
@@ -152,8 +152,10 @@ function match(re::Regex, str::UTF8String, idx::Integer, add_opts::UInt32=UInt32
     RegexMatch(mat, cap, ovec[1]+1, off, re)
 end
 
+_utf8(str) = utf8(str)
```

Comment:
I take your second point though. Although I think instead of narrowing the signature of `match`, it might make  more sense to widen it so you could pass in any subject string that has `utf8` defined on it.